crates/api/src/agent.rs

```diff
@@ -83,11 +83,11 @@ use std::sync::Arc;
 /// Defines a type capable of cryptographic signatures.
 pub trait Signer {
     /// Sign the encoded data, returning the resulting detached signature bytes.
-    fn sign(
-        &self,
-        agent_info: &AgentInfo,
-        message: &[u8],
-    ) -> BoxFut<'_, K2Result<bytes::Bytes>>;
+    fn sign<'a, 'b: 'a, 'c: 'a>(
```

Comment:
If you used a `'static` `b""` type in calling this message, it would pollute the requirements and make it seem as if the reference to self also had to be `'static`. As far as I can tell, this now says `'b` and `'c` can be different than the `'a` used in self, but they have to be valid at least as long as self.

Although I still find lifetimes pretty fiddly...